pkgdb/src/buildenv/command.cc

```diff
@@ -86,6 +91,27 @@ BuildEnvCommand::run()
   auto store = this->getStore();
   auto state = this->getState();
 
+  if ( this->storePath.has_value() && this->outLink.has_value() )
```

Comment:
question nonblocking: so is this basically just reimplementing `nix-store --add-root outLink --realise storePath`? Is there an advantage to doing that rather than just exec'ing nix?

Reply:
Yeah, that's what it's doing. The benefit is that we can keep the number of `nix` invocations minimal since at some point we may decide not to include the executable itself. I also don't think we currently have any other `nix` invocations in `flox` right now, so going from zero to one feels worse. This is also basically the same amount of code as exec'ing `nix`.

Reply:
suggestion nonblocking: add comment like `This essentially reimplements 'nix-store --add-root outLink --realise', but we do so to have the flexibility of not depending on the nix binary`

Reply:
Probably not going to do that, we're doing a bunch of stuff that's reimplementing Nix functionality so it doesn't make sense to me to leave comments about it.